src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -603,6 +603,51 @@ void CallGenerator::do_late_inline_helper() {
     C->remove_macro_node(call);
   }
 
+  if (callprojs.resproj != NULL && call->is_CallStaticJava() &&
```

Comment:
Please extract the new code into a method with a descriptive name.

Reply:
OK. I will refactor these codes ASAP.